loadbalancer/algorithm.go

```diff
@@ -72,7 +76,10 @@ func newRandom(endpoints []string) routing.LBAlgorithm {
 
 // Apply implements routing.LBAlgorithm with a stateless random algorithm.
 func (r *random) Apply(ctx *routing.LBContext) routing.LBEndpoint {
-	return ctx.Route.LBEndpoints[r.rand.Intn(len(ctx.Route.LBEndpoints))]
+	r.mutex.Lock()
+	index := r.rand.Intn(ctx.Route.LBEndpoints.Length())
+	r.mutex.Unlock()
```

Comment:
I think unlock should be in done in defer otherwise mutex will stay locked if `Intn` panics.
`Intn` panics if argument is <= 0, here it assumes `LBEndpoints.Length()` is never zero.

Reply:
Done.